Sming/Components/ssl/include/Network/Ssl/Certificate.h

```diff
@@ -65,28 +66,31 @@ class Certificate
 			MAX
 	};
 
-	/**
-	 * @brief Obtain a string describing the given name component
-	 */
-	static String getRdnTypeString(Certificate::RDN rdn);
-
 	virtual ~Certificate()
 	{
 	}
 
 	/**
-	 * @brief Check if certificate fingerprint (SHA1) matches the one given.
-	 * @param hash - SHA1 fingerprint to match against
-	 * @retval bool true on match, false otherwise
+	 * @name Obtain certificate fingerprint
+	 * @param fingerprint On success, returned fingerprint
+	 * @retval bool true on success, false if fingerprint /not available
+	 * @{
 	 */
-	virtual bool matchFingerprint(const uint8_t* hash) const = 0;
+	virtual bool getFingerprint(Fingerprint::Cert::Sha1& fingerprint) const
```

Comment:
Unfortunately, c++17 is not the default yet :(
If it were, this interface could have been simplified to
```cpp
virtual std::optional<Fingerprint::Cert::Sha1> getFingerprint() const
```
(A advantage of return-by-value over by-reference parameters is that the returned type does not need a default constructor, nor does it have to support any kind of 'invalid state', which is usually achieved by the default constructor)